src/biz/collab/ops.rs

```diff
@@ -137,3 +151,79 @@ pub async fn get_collab_member_list(
   let collab_member = database::collab::select_collab_members(&params.object_id, pg_pool).await?;
   Ok(collab_member)
 }
+
+pub async fn get_user_workspace_structure(
+  group_manager: AppStateGroupManager,
+  collab_storage: Arc<CollabAccessControlStorage>,
+  uid: i64,
+  workspace_id: String,
+  depth: u32,
+) -> Result<FolderView, AppError> {
+  let depth_limit = 10;
+  if depth > depth_limit {
+    return Err(AppError::InvalidRequest(format!(
+      "Depth {} is too large (limit: {})",
+      depth, depth_limit
+    )));
+  }
+  let folder = get_latest_collab_folder(group_manager, collab_storage, &uid, &workspace_id).await?;
+  let folder_view = FolderView::from_folder(&folder, depth);
+  Ok(folder_view)
+}
+
+pub async fn get_latest_collab_folder(
+  group_manager: AppStateGroupManager,
+  collab_storage: Arc<CollabAccessControlStorage>,
+  uid: &i64,
+  workspace_id: &str,
+) -> Result<Folder, AppError> {
+  let encoded_collab = get_latest_collab_encoded(
+    group_manager,
+    collab_storage,
+    uid,
+    workspace_id,
+    workspace_id,
+    CollabType::Folder,
+  )
+  .await?;
+  let folder = Folder::from_collab_doc_state(
+    uid,
+    CollabOrigin::Server,
+    DataSource::DocStateV1(encoded_collab.doc_state.to_vec()),
```

Comment:
please review all the places that using encode_collab. You can use into() instead
<img width="752" alt="image" src="https://github.com/user-attachments/assets/7e373dcf-de97-4ead-a215-293d8ef0012a">

Reply:
done